include/xtypes/DynamicData.hpp

```diff
@@ -864,6 +866,196 @@ class DynamicData : public WritableDynamicDataRef
     /// \returns a WritableDynamicDataRef identifying th DynamicData.
     WritableDynamicDataRef ref() const { return WritableDynamicDataRef(*this); }
+
+    template<typename T, class = std::is_arithmetic<T>>
+    T cast() const
+    {
+        switch (type_.kind())
+        {
+            case TypeKind::BOOLEAN_TYPE:
+            {
+                bool temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::INT_8_TYPE:
+            {
+                int8_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::UINT_8_TYPE:
+            {
+                uint8_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::INT_16_TYPE:
+            {
+                int16_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::UINT_16_TYPE:
+            {
+                uint16_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::INT_32_TYPE:
+            {
+                int32_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::UINT_32_TYPE:
+            {
+                uint32_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::INT_64_TYPE:
+            {
+                int64_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::UINT_64_TYPE:
+            {
+                uint64_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::FLOAT_32_TYPE:
+            {
+                float temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::FLOAT_64_TYPE:
+            {
+                double temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::FLOAT_128_TYPE:
+            {
+                long double temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::CHAR_8_TYPE:
+            {
+                char temp = *this;
+                return static_cast<T>(temp);
+            }
+            case TypeKind::CHAR_16_TYPE:
+            {
+                wchar_t temp = *this;
+                return static_cast<T>(temp);
+            }
+            //case TypeKind::ENUMERATION_TYPE:
+            /*
+            case TypeKind::STRING_TYPE:
+            {
+                std::string temp = *this;
+                return reinterpret_cast<T>(temp);
+            }
+            case TypeKind::WSTRING_TYPE:
+            {
+                std::wstring temp = *this;
+                return reinterpret_cast<T>(temp);
+            }
+            */
+        }
+        return T();
+    }
+
+    template<typename T = std::string>
+    T cast() const
```

Comment:
I think that can be retrive in one line if you use the instance and perform an static_cast before call to std::to_string()

Reply:
Comments already addressed.